perses/distributed/feptasks.py

```diff
@@ -271,7 +275,11 @@ def run_equilibrium(thermodynamic_state, sampler_state, mc_move, topology, n_ite
     reduced_potential_final_frame : float
         Unitless reduced potential (kT) of the final frame of the trajectory
     """
-    n_atoms = topology.n_atoms
+    #get the atom indices we need to subset the topology and positions
+    atom_indices = topology.select('not water')
```

Comment:
Just take atom indices as an argument (defaulting to `None`, which means all atoms).